clientv3/integration/client_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/lease"
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+func v3Put(clus *integration.ClusterV3, key, val string) (*clientv3.PutResponse, error) {
+	kvc := clientv3.NewKV(clus.RandClient())
+	resp, err := kvc.Put(key, val, lease.NoLease)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func v3Range(clus *integration.ClusterV3, key string) (*clientv3.RangeResponse, error) {
+	kvc := clientv3.NewKV(clus.RandClient())
+	resp, err := kvc.Range(key, "", 0, 0, nil)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func TestPutOverwrite(t *testing.T) {
```

Comment:
let's remove this test. we should focus on testing client, not server. we just need to ensure the client works correctly. we do not need to test the server workflow.